resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/StringParameterInjector.java

```diff
@@ -80,40 +80,39 @@ protected void initialize(Class type, Type genericType, String paramName, Class
       baseType = type;
       baseGenericType = genericType;
 
-      if (type.isArray()) baseType = type.getComponentType();
-      if (List.class.isAssignableFrom(type))
-      {
-         isCollection = true;
-         collectionType = ArrayList.class;
-      }
-      else if (SortedSet.class.isAssignableFrom(type))
-      {
-         isCollection = true;
-         collectionType = TreeSet.class;
-      }
-      else if (Set.class.isAssignableFrom(type))
+      if (type.isArray())
       {
-         isCollection = true;
-         collectionType = HashSet.class;
+    	  baseType = type.getComponentType();
       }
-      if (isCollection)
+     
+      if (!baseType.isPrimitive())
       {
-         if (genericType != null && genericType instanceof ParameterizedType)
+         paramConverter = factory.getParamConverter(baseType, baseGenericType, annotations);
+         if (paramConverter != null) 
          {
-            ParameterizedType zType = (ParameterizedType) genericType;
-            baseType = Types.getRawType(zType.getActualTypeArguments()[0]);
-            baseGenericType = zType.getActualTypeArguments()[0];

```

Comment:
>    
      if (!baseType.isPrimitive())
      {
         //This call is ok baseType is not a primitive
         paramConverter = factory.getParamConverter(baseType, baseGenericType, annotations);
         ...
	         if (genericType != null && genericType instanceof ParameterizedType)
	         {
	            ParameterizedType zType = (ParameterizedType) baseGenericType;
                // baseType is modified here so it can become a primitive (seem's technicaly impossible to me in this case but we never know...)
	            baseType = Types.getRawType(zType.getActualTypeArguments()[0]);
	            baseGenericType = zType.getActualTypeArguments()[0];
	         }
          ...
             // If baseType has been modified to a primitive it's KO, we change the behavior of the original code
	         paramConverter = factory.getParamConverter(baseType, baseGenericType, annotations);
	      ...
       }